groups/client.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+
+	admin "google.golang.org/api/admin/directory/v1"
+	groupssettings "google.golang.org/api/groupssettings/v1"
+	"google.golang.org/api/option"
+)
+
+type AdminServiceClient interface {
+	GetGroup(groupKey string) (*admin.Group, error)
```

Comment:
is it possible to consume this code vs. copying it?

Reply:
yup, dropped the go code. Makefile builds a binary now.

get works but install doesn't :( 

```
go install k8s.io/k8s.io/groups@v0.0.0-20220124221514-b894a331a999: k8s.io/k8s.io/groups@v0.0.0-20220124221514-b894a331a999
        The go.mod file for the module providing named packages contains one or
        more replace directives. It must not contain directives that would cause
        it to be interpreted differently than if it were the main module.
make: *** [run] Error 1
```

Reply:
On a second look, I don't think this will be possible. 

https://github.com/kubernetes/k8s.io/blob/main/groups/groups_test.go is doing critical validation.

Reply:
cool no worries